erdjs/src/dapp/walletConnectProvider.ts

```diff
@@ -0,0 +1,196 @@
+            await this.walletConnector?.createSession({ chainId }).then(() => {
+                const uri = this.walletConnector?.uri;
+                if (uri !== undefined) {
+                    returnUrl = uri;
+                }
+            });
+        }
+        return returnUrl;
+    }
+
+    loginAccount = (address: string) => {
+        if (this.addressIsValid(address)) {
+            this.address = address;
+            this.dispatchEvent(this.onWalletConnectLogin);
+        } else {
+            if (this.walletConnector?.connected) {
+                this.walletConnector?.killSession();
+            }
+        }
+    };
+
+    onConnect = async (error: any, { params }: any) => {
+        if (error) {
+            throw error;
+        }
+        if (!params || !params[0]) {
+            Logger.error("Wallet Connect missing payload");
+            throw new Error("missing payload");
+        }
+        const {
+            accounts: [account],
+        } = params[0];
+
+        this.loginAccount(account);
+    };
+
+    onDisconnect = async (error: any) => {
+        if (error) {
+            throw error;
+        }
+
+        this.dispatchEvent(this.onWalletConnectDisconect);
+    };
+
+    /**
+     * Mocks a logout request by returning true
+     */
+    async logout(): Promise<boolean> {
+        if (!this.walletConnector) {
+            Logger.error("logout: Wallet Connect not initialised, call init() first");
+            throw new Error("Wallet Connect not initialised, call init() first");
+        }
+        this.walletConnector.killSession();
+        return true;
+    }
+
+    /**
+     * Fetches current selected ledger address
+     */
+    async getAddress(): Promise<string> {
+        return this.address;
+    }
+
+    /**
+     * Signs and sends a transaction. Returns the transaction hash
+     * @param transaction
+     */
+    async sendTransaction(transaction: Transaction): Promise<Transaction> {
+        if (!this.walletConnector) {
+            Logger.error("sendTransaction: Wallet Connect not initialised, call init() first");
+            throw new Error("Wallet Connect not initialised, call init() first");
+        }
+
+        const address = await this.getAddress();
+        const sig = await this.signTransaction("erd_sign", this.prepareWalletConnectMessage(transaction, address));
+        if (sig && sig.signature) transaction.applySignature(new Signature(sig.signature), new Address(address));
+
+        await transaction.send(this.provider);
+
+        return transaction;
+    }
+
+    private async signTransaction(method: any, params: any): Promise<any> {
+        if (!this.walletConnector) {
+            Logger.error("signTransaction: Wallet Connect not initialised, call init() first");
+            throw new Error("Wallet Connect not initialised, call init() first");
+        }
+        if (this.walletConnector.connected) {
+            const response = await this.walletConnector.sendCustomRequest({
```

Comment:
You can return directly `return await this.walletConnector.sendC....`. `response` constant is redundant

Reply:
done